exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/AnalyzeTableHandler.java

```diff
@@ -0,0 +1,256 @@
+import org.apache.drill.exec.store.dfs.NamedFormatPluginConfig;
+import org.apache.drill.exec.store.parquet.ParquetFormatConfig;
+import org.apache.drill.exec.util.Pointer;
+import org.apache.drill.exec.work.foreman.ForemanSetupException;
+import org.apache.drill.exec.work.foreman.SqlUnsupportedException;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.FileStatus;
+
+public class AnalyzeTableHandler extends DefaultSqlHandler {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(AnalyzeTableHandler.class);
+
+  public AnalyzeTableHandler(SqlHandlerConfig config, Pointer<String> textPlan) {
+    super(config, textPlan);
+  }
+
+  @Override
+  public PhysicalPlan getPlan(SqlNode sqlNode)
+      throws ValidationException, RelConversionException, IOException, ForemanSetupException {
+    final SqlAnalyzeTable sqlAnalyzeTable = unwrap(sqlNode, SqlAnalyzeTable.class);
+
+    verifyNoUnsupportedFunctions(sqlAnalyzeTable);
+
+    SqlIdentifier tableIdentifier = sqlAnalyzeTable.getTableIdentifier();
+    SqlSelect scanSql = new SqlSelect(
+        SqlParserPos.ZERO,              /* position */
+        SqlNodeList.EMPTY,              /* keyword list */
+        getColumnList(sqlAnalyzeTable), /* select list */
+        tableIdentifier,                /* from */
+        null,                           /* where */
+        null,                           /* group by */
+        null,                           /* having */
+        null,                           /* windowDecls */
+        null,                           /* orderBy */
+        null,                           /* offset */
+        null                            /* fetch */
+    );
+
+    final ConvertedRelNode convertedRelNode = validateAndConvert(rewrite(scanSql));
+    final RelNode relScan = convertedRelNode.getConvertedNode();
+    final String tableName = sqlAnalyzeTable.getName();
+    final AbstractSchema drillSchema = SchemaUtilites.resolveToDrillSchema(
+        config.getConverter().getDefaultSchema(), sqlAnalyzeTable.getSchemaPath());
+    Table table = SqlHandlerUtil.getTableFromSchema(drillSchema, tableName);
+
+    if (table == null) {
+      throw UserException.validationError()
+          .message("No table with given name [%s] exists in schema [%s]", tableName,
+              drillSchema.getFullSchemaName())
+          .build(logger);
+    }
+
+    if(! (table instanceof DrillTable)) {
+      return notSupported(tableName);
+    }
+
+    if (table instanceof DrillTable) {
+      DrillTable drillTable = (DrillTable) table;
+      final Object selection = drillTable.getSelection();
+      if (!(selection instanceof FormatSelection)) {
+        return notSupported(tableName);
+      }
+      // Do not support non-parquet tables
+      FormatSelection formatSelection = (FormatSelection) selection;
+      FormatPluginConfig formatConfig = formatSelection.getFormat();
+      if (!((formatConfig instanceof ParquetFormatConfig)
```

Comment:
It would be better to add an API to GroupScan interface for checking if the underlying GroupScan supports statistics collection.   For now, only ParquetGroupScan will return true for that.